test/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionControllerSpec.scala

```diff
@@ -399,6 +399,63 @@ class PersonalDetailsCollectionControllerSpec extends UnitSpec with MockFactory
     }
   }
 
+  "showHaveYourNationalInsuranceNumber" should {
+    "return OK with the ability to confirm whether you have your Nino" in new Setup {
+
+      val req: FakeRequest[AnyContentAsEmpty.type] = request.withSession(
+        "firstName" -> "Jim",
+        "lastName" -> "Ferguson",
+        "dob" -> "1939-09-01"
+      )
+      val result: Future[Result] = controller.showHaveYourNationalInsuranceNumber(completionUrl, failureUrl)(req)
+
+      contentType(result) shouldBe Some(HTML)
+      charset(result) shouldBe Some("utf-8")
+
+      val document: Document = Jsoup.parse(contentAsString(result))
+
+      document.select("main span[class='govuk-caption-l']").text() shouldBe messages("do_you_have_your_nino.h2")
+      document.select("h1").text() shouldBe messages("do_you_have_your_nino.h1")
+      document.select("main p").get(0).text() shouldBe messages("do_you_have_your_nino.p1")
+      document.select("main p").get(1).text() shouldBe messages("do_you_have_your_nino.p2")
+      document.select("main p").get(2).text() shouldBe messages("do_you_have_your_nino.p3")
+      document.select("main li").get(0).text() shouldBe messages("do_you_have_your_nino.li1")
+      document.select("main li").get(1).text() shouldBe messages("do_you_have_your_nino.li2")
+      document.select("main li").get(2).text() shouldBe messages("do_you_have_your_nino.li3")
+      document.select("details summary").text() shouldBe messages("do_you_have_your_nino.details")
```

Comment:
Something else I've just thought of, should these tests go into a view spec rather than the controller spec?
In FPR we have the view tests, and the the controller tests can focus on the logic in the controller, check that the controller routes to the correct page etc.

Reply:
Seeing as this repo barely has view specific tests, I'd be tempted to leave as-is, but wondering what you think

Reply:
I did originally start with the approach of a separate viewSpec.. however it was not possible to trim down the setup as much as I wanted because of the need to send in all the other templates to PersonalDetailsCollectionController, so I decided to keep it in the one file to avoid duplicating a lot of code.